src/meta/src/stream/stream_manager.rs

```diff
@@ -12,60 +12,52 @@
 use risingwave_common::catalog::TableId;
-use risingwave_common::hash::{ActorMapping, ParallelUnitMapping};
 use risingwave_pb::catalog::Table;
-use risingwave_pb::common::{ActorInfo, Buffer, WorkerType};
-use risingwave_pb::meta::table_fragments::actor_status::ActorState;
-use risingwave_pb::meta::table_fragments::fragment::FragmentDistributionType;
-use risingwave_pb::meta::table_fragments::ActorStatus;
-use risingwave_pb::stream_plan::stream_node::NodeBody;
-use risingwave_pb::stream_plan::{Dispatcher, DispatcherType, StreamNode};
+use risingwave_pb::common::ActorInfo;
+use risingwave_pb::stream_plan::Dispatcher;
 use risingwave_pb::stream_service::{
     BroadcastActorInfoTableRequest, BuildActorsRequest, HangingChannel, UpdateActorsRequest,
 };
 use uuid::Uuid;
 
-use super::ScheduledLocations;
+use super::Locations;
 use crate::barrier::{BarrierScheduler, Command};
 use crate::hummock::HummockManagerRef;
-use crate::manager::{
-    ClusterManagerRef, FragmentManagerRef, FragmentVNodeInfo, MetaSrvEnv, WorkerId,
-};
-use crate::model::{ActorId, FragmentId, TableFragments};
+use crate::manager::{ClusterManagerRef, FragmentManagerRef, MetaSrvEnv};
+use crate::model::{ActorId, TableFragments};
 use crate::storage::MetaStore;
-use crate::stream::{Scheduler, SourceManagerRef};
+use crate::stream::SourceManagerRef;
 use crate::MetaResult;
 
 pub type GlobalStreamManagerRef<S> = Arc<GlobalStreamManager<S>>;
 
 /// [`CreateStreamingJobContext`] carries one-time infos.
-#[derive(Default)]
+///
+/// Note: for better readability, keep this struct complete and immutable once created.
+#[cfg_attr(test, derive(Default))]
 pub struct CreateStreamingJobContext {
     /// New dispatchers to add from upstream actors to downstream actors.
     pub dispatchers: HashMap<ActorId, Vec<Dispatcher>>,
-    /// Upstream mview actor ids grouped by worker node id.
-    pub upstream_worker_actors: HashMap<WorkerId, HashSet<ActorId>>,
+
     /// Upstream mview actor ids grouped by table id.
-    pub table_mview_map: HashMap<TableId, Vec<ActorId>>,
-    /// Dependent table ids
-    pub dependent_table_ids: HashSet<TableId>,
-    /// Internal TableID to Table mapping
+    pub upstream_mview_actors: HashMap<TableId, Vec<ActorId>>,
+
+    /// Internal tables in the streaming job.
     pub internal_tables: HashMap<u32, Table>,
-    /// The upstream tables of all fragments containing chain nodes.
-    /// These fragments need to be colocated with their upstream tables.
-    ///
-    /// They are scheduled in `resolve_chain_node`.
-    pub chain_fragment_upstream_table_map: HashMap<FragmentId, TableId>,
-    /// The SQL definition of this streaming job. Used for debugging only.
-    pub streaming_definition: String,
 
+    /// The locations of the actors to build in the streaming job.
+    pub building_locations: Locations,
+
+    /// The locations of the existing actors, essentially the upstream mview actors to update.
+    pub existing_locations: Locations,
```

Comment:
This is derived from the `ActorGraphBuilder`, and can be directly used for RPCs like building hanging channels or broadcast actor info.